### PR TITLE
vss: fix DeleteSnapshots() and GetSnapshotProperties() on 386

### LINKED ISSUE #3087: --use-fs-snapshot failed on windows/386
Output of `restic version`
--------------------------
restic 0.11.0 compiled with go1.15.2 on windows/386

How did you run restic exactly?
-------------------------------
restic --use-fs-snapshot d:\test

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
local filesystem repository

Expected behavior
-----------------
VSS don't failed

Actual behavior
---------------
~~~
repository 43e035ae opened successfully, password is correct
creating VSS snapshot for [d:\]
error: failed to create snapshot for [d:\]: VSS error: GetSnapshotProperties() failed: E_INVALIDARG (0x80070057)
.....
~~~

Steps to reproduce the behavior
-------------------------------
Use `windows/386` restic version on 32-bit Windows with `--use-fs-snapshot` flag.

Do you have any idea what may have caused this?
-----------------------------------------------
Nobody uses 32-bit Windows, and syscalls from `vss_windows.go` are hardly tested.
I was absolute sure this bug was fixed by @fgma, but this piece of code seems to be cursed.

Do you have an idea how to solve the issue?
-------------------------------------------
```diff
diff --git a/internal/fs/vss_windows.go b/internal/fs/vss_windows.go
index e24dade..8cee09f 100644
--- a/internal/fs/vss_windows.go
+++ b/internal/fs/vss_windows.go
@@ -482,7 +482,7 @@ func (vss *IVssBackupComponents) DeleteSnapshots(snapshotID ole.GUID) (int32, ol
        var result uintptr = 0
 
        if runtime.GOARCH == "386" {
-               id := (*[4]uintptr)(unsafe.Pointer(ole.IID_NULL))
+               id := (*[4]uintptr)(unsafe.Pointer(&snapshotID))
 
                result, _, _ = syscall.Syscall9(vss.getVTable().deleteSnapshots, 9,
                        uintptr(unsafe.Pointer(vss)), id[0], id[1], id[2], id[3],
@@ -506,7 +506,7 @@ func (vss *IVssBackupComponents) GetSnapshotProperties(snapshotID ole.GUID,
        var result uintptr = 0
 
        if runtime.GOARCH == "386" {
-               id := (*[4]uintptr)(unsafe.Pointer(ole.IID_NULL))
+               id := (*[4]uintptr)(unsafe.Pointer(&snapshotID))
 
                result, _, _ = syscall.Syscall6(vss.getVTable().getSnapshotProperties, 6,
                        uintptr(unsafe.Pointer(vss)), id[0], id[1], id[2], id[3],
```

I think restic needs integration tests with **real** testing of VSS. Perhaps carefully prepared VHD image with many partitions of different types, broken mount points and other edge cases.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes! Yes!

### PR DESCRIPTION
**Pull request by fgma**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Creating VSS snapshots doesn't work on 386. Fix proposed by @DRON-666. For details see https://github.com/restic/restic/issues/3087.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Closes #3087

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [ ] I have added tests for all changes in this PR
- [ ] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

**Comment by fgma** [comment]
Failed CI seems to be unrelated to the PR.

**Comment by fd0** [comment]
It is, maybe it works if you rebase on master (which I think I've fixed yesterday).

**Review by fd0 — APPROVED** [review]
LGTM! We can merge this once the CI is happy.

File: changelog/unreleased/issue-3090 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Bugfix: --use-fs-snapshot failed on windows/386
```
Comment (greatroar) [inline changelog/unreleased/issue-3090:1]: Bugfix: --use-fs-snapshot now works on windows/386

File: changelog/unreleased/issue-3090 [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+restic failed to create VSS snapshots on windows/386 because of invalid API calls.
```
Comment (greatroar) [inline changelog/unreleased/issue-3090:3]: "because of invalid API calls" is an implementation detail. I'd include the actual error message that users would have seen, so they can search for it. Also, I think @rawtaz wants Restic with a capital letter at the start of a sentence and you forgot to mention that the issue is fixed now :)

Comment (rawtaz) [reply changelog/unreleased/issue-3090:3]: You read my mind :D

Comment (DRON-666) [reply changelog/unreleased/issue-3090:3]: >I'd include the actual error message that users would have seen, so they can search for it.

It's not a good idea: GetSnapshotProperties failed with the same error message for pretty much [everything](https://github.com/restic/restic/pull/2274#issuecomment-683151351).

File: changelog/unreleased/issue-3090 (hunk 1 above)
Comment (fgma) [reply changelog/unreleased/issue-3090]: Fixed in https://github.com/restic/restic/pull/3090/commits/88208c3db27da33ab9141f10abf384b7a9045221

File: changelog/unreleased/issue-3090 (hunk 2 above)
Comment (fgma) [reply changelog/unreleased/issue-3090]: Fixed in https://github.com/restic/restic/pull/3090/commits/88208c3db27da33ab9141f10abf384b7a9045221

**Comment by fd0** [comment]
Thanks!
